translator/trace/protospan_to_jaegerthrift.go

```diff
@@ -0,0 +1,360 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracetranslator
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/jaegertracing/jaeger/thrift-gen/jaeger"
+
+	commonpb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
+	agenttracepb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1"
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+)
+
+// OCProtoToJaegerThrift translates OpenCensus trace data into the Jaeger Thrift format.
+func OCProtoToJaegerThrift(ocBatch *agenttracepb.ExportTraceServiceRequest) (*jaeger.Batch, error) {
+	if ocBatch == nil {
+		return nil, nil
+	}
+
+	jSpans, err := ocSpansToJaegerSpans(ocBatch.Spans)
+	if err != nil {
+		return nil, err
+	}
+
+	jb := &jaeger.Batch{
+		Process: ocNodeToJaegerProcess(ocBatch.Node),
+		Spans:   jSpans,
+	}
+
+	return jb, nil
+}
+
+func ocNodeToJaegerProcess(node *commonpb.Node) *jaeger.Process {
+	if node == nil {
+		return nil
+	}
+
+	var jTags []*jaeger.Tag
+	nodeAttribsLen := len(node.Attributes)
+	if nodeAttribsLen > 0 {
+		jTags = make([]*jaeger.Tag, 0, nodeAttribsLen)
+		for k, v := range node.Attributes {
+			str := v
+			jTag := &jaeger.Tag{
+				Key:   k,
+				VType: jaeger.TagType_STRING,
+				VStr:  &str,
+			}
+			jTags = append(jTags, jTag)
+		}
+	}
+
+	if node.Identifier != nil && node.Identifier.HostName != "" {
```

Comment:
Cool! How about Pid and StartTimestamp?

Field|Reference
---|---
Pid|[ProcessIdentifier.Pid](https://godoc.org/github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1#ProcessIdentifier.Pid)
StartTimestamp|[ProcessIdentifier.StartTimestamp](https://godoc.org/github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1#ProcessIdentifier.StartTimestamp)

Reply:
Added.